src/coreclr/jit/codegenlinear.cpp

```diff
@@ -1223,6 +1223,13 @@ void CodeGen::genUnspillRegIfNeeded(GenTree* tree)
             {
                 spillType = lclLoadType;
             }
+
+#if defined(TARGET_LOONGARCH64)
+            if (varTypeIsFloating(spillType) && emitter::isGeneralRegister(tree->GetRegNum()))
```

Comment:
One way to interpret this code is to mean we now have `LCL_VAR<V00 float> int` nodes in the system.

Where do they come from?

Reply:
On LoongArch64's ABI, Float type arg is passed by integer register while there is no float register left.

Reply:
Yes, but that doesn't explain it fully.

For the caller (method being compiled), we would have `LCL_VAR V00<float> float`, i. e. both the local variable have `TYP_FLOAT` and the node. And obviously a `TYP_FLOAT` node cannot have an integer register assigned to it.

So is it the case for callees then? I. e. `fgMorphArgs` retyping?

Could you please share a `JitDump` where we will see this case?

Reply:
In test cases this is ok.

I will add some asm tomorrow.